File: scripts/batch-generate.py
"""
FindInChina 批量内容生成脚本（Python 版）
用 urllib 直接调 DeepSeek API，避免 node 依赖问题
输出到 generated-content/ 目录
"""
import json
import os
import time
import urllib.request
import urllib.error

API_KEY = "sk-fa9d6a4dd4ba4d29b98606509f1bd601"
API_URL = "https://api.deepseek.com/chat/completions"
MODEL = "deepseek-v4-flash"

OUT_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "generated-content")


def call_deepseek(prompt, max_tokens=6000, temperature=0.5, json_mode=True):
    payload = {
        "model": MODEL,
        "messages": [
            {"role": "system", "content": "You are a meticulous brand researcher producing structured JSON data."},
            {"role": "user", "content": prompt},
        ],
        "temperature": temperature,
        "max_tokens": max_tokens,
    }
    if json_mode:
        payload["response_format"] = {"type": "json_object"}

    req = urllib.request.Request(
        API_URL,
        data=json.dumps(payload).encode("utf-8"),
        headers={
            "Content-Type": "application/json",
            "Authorization": f"Bearer {API_KEY}",
        },
    )
    with urllib.request.urlopen(req, timeout=120) as resp:
        data = json.loads(resp.read().decode("utf-8"))
    return data["choices"][0]["message"]["content"]


REPORT_PROMPT = """You are the lead researcher at FindInChina, a curated editorial platform helping global buyers source verified Chinese brands.

Create a "FindIn Report" for the category: **{category}**

Brands to cover: {brands}

Output MUST be valid JSON with EXACTLY this structure (no markdown, no code fences):
{{
  "slug": "kebab-case-unique-slug",
  "title": "Compelling headline with brand count and category",
  "subtitle": "One-line teaser naming the brands",
  "category": "{category}",
  "brand_count": 5,
  "criteria_count": 4,
  "read_time_min": 8,
  "last_verified": "2026-08-17",
  "excerpt": "One-line summary for list cards",
  "intro_text": "3-4 paragraph editorial intro",
  "criteria_text": ["Real export operation", "Working warranty", "Product line currency", "Verifiable contact"],
  "brand_blocks": [
    {{
      "brand_name": "Brand English Name",
      "slug": "brand-slug",
      "quick_take": "One sentence: who this is best for",
      "why_vetted": ["3-4 specific vetted facts with data"],
      "who_for": ["2-3 buyer personas"],
      "expert_quote": "One strong editorial quote",
      "gallery": []
    }}
  ],
  "comparison_rows": [
    {{
      "brand_slug": "brand-slug",
      "brand_name": "Brand",
      "price": 999,
      "price_label": "$999",
      "warranty": "2 yr",
      "countries": 5,
      "rating": 4.0,
      "third_party": "\u2014"
    }}
  ],
  "methodology_text": "2 paragraphs explaining the 4-criteria scoring method",
  "sources": ["source 1", "source 2"]
}}

IMPORTANT:
- Use REAL brand names, REAL approximate prices, REAL warranty terms (research from official sites)
- Editorial tone: vetted, trustworthy, specific data \u2014 NOT generic filler
- At least 4-5 brands in brand_blocks
- All text in English, written for global buyers sourcing from China
- Do NOT include speculative data"""


def generate_report(category, brands):
    prompt = REPORT_PROMPT.format(category=category, brands=", ".join(brands))
    raw = call_deepseek(prompt, max_tokens=6000, temperature=0.5)
    try:
        return json.loads(raw)
    except json.JSONDecodeError:
        # 尝试从代码块中提取
        import re
        match = re.search(r"```json\s*([\s\S]*?)```", raw)
        if match:
            return json.loads(match.group(1))
        print(f"[WARN] JSON 解析失败: {raw[:300]}")
        return None


def main():
    os.makedirs(OUT_DIR, exist_ok=True)
    reports = [
        ("Outdoor gear", ["Naturehike", "KingCamp", "FIELDOOR", "DOD", "3F UL Gear"]),
        ("GaN chargers", ["CUKTECH", "PISEN", "Baseus", "UGREEN", "Mcdodo"]),
        ("Mechanical keyboards", ["Keychron", "Epomaker", "Akko", "Royal Kludge", "Aula"]),
    ]

    summary = {"generated_at": time.strftime("%Y-%m-%d %H:%M:%S"), "reports": []}

    for category, brands in reports:
        print(f"生成报告: {category}...", flush=True)
        try:
            data = generate_report(category, brands)
            if data:
                slug = data.get("slug") or category.lower().replace(" ", "-")
                filepath = os.path.join(OUT_DIR, f"report-{slug}.json")
                with open(filepath, "w", encoding="utf-8") as f:
                    json.dump(data, f, ensure_ascii=False, indent=2)
                summary["reports"].append({"category": category, "slug": slug, "title": data.get("title"), "file": filepath})
                print(f"  OK: {data.get('title')}", flush=True)
            else:
                summary["reports"].append({"category": category, "error": "parse failed"})
        except Exception as e:
            summary["reports"].append({"category": category, "error": str(e)})
            print(f"  ERROR: {e}", flush=True)
        time.sleep(2)

    with open(os.path.join(OUT_DIR, "summary.json"), "w", encoding="utf-8") as f:
        json.dump(summary, f, ensure_ascii=False, indent=2)
    print(f"\n=== 完成: {len(summary['reports'])} 篇报告 ===", flush=True)
    print(f"输出目录: {OUT_DIR}", flush=True)


if __name__ == "__main__":
    main()

File: scripts/batch-generate.mjs
/**
 * FindInChina 批量内容生成脚本
 * ============================
 * 用 DeepSeek 生成品牌调研报告，先输出到本地文件（不依赖数据库）
 * 等 Supabase secret key 修复后，可将数据灌入 findin_reports 表
 */
import OpenAI from 'openai'
import fs from 'fs'
import path from 'path'

const client = new OpenAI({
  apiKey: 'sk-fa9d6a4dd4ba4d29b98606509f1bd601',
  baseURL: 'https://api.deepseek.com',
})

const OUT_DIR = path.join(process.cwd(), 'generated-content')

// 报告生成 Prompt —— 严格匹配 findin_reports 表结构
const REPORT_PROMPT = (category, brands) => `You are the lead researcher at FindInChina, a curated editorial platform helping global buyers source verified Chinese brands.

Create a "FindIn Report" for the category: **${category}**

Brands to cover: ${brands.join(', ')}

Output MUST be valid JSON with EXACTLY this structure (no markdown, no code fences):
{
  "slug": "kebab-case-unique-slug",
  "title": "Compelling headline with brand count and category",
  "subtitle": "One-line teaser naming the brands",
  "category": "${category}",
  "brand_count": number,
  "criteria_count": 4,
  "read_time_min": 8,
  "last_verified": "2026-08-17",
  "excerpt": "One-line summary for list cards",
  "intro_text": "3-4 paragraph editorial intro (no markdown headers)",
  "criteria_text": ["Real export operation", "Working warranty", "Product line currency", "Verifiable contact"],
  "brand_blocks": [
    {
      "brand_name": "Brand English Name",
      "slug": "brand-slug",
      "quick_take": "One sentence: who this is best for",
      "why_vetted": ["3-4 specific vetted facts with data"],
      "who_for": ["2-3 buyer personas"],
      "expert_quote": "One strong editorial quote",
      "gallery": []
    }
  ],
  "comparison_rows": [
    {
      "brand_slug": "brand-slug",
      "brand_name": "Brand",
      "price": 999,
      "price_label": "$999",
      "warranty": "2 yr",
      "countries": 5,
      "rating": 4.0,
      "third_party": "—"
    }
  ],
  "methodology_text": "2 paragraphs explaining the 4-criteria scoring method",
  "sources": ["source 1", "source 2"]
}

IMPORTANT:
- Use REAL brand names, REAL approximate prices, REAL warranty terms (research from official sites)
- Editorial tone: vetted, trustworthy, specific data — NOT generic filler
- Brand blocks must have at least 3 brands, ideally 4-5
- All text in English, written for global buyers sourcing from China
- Do NOT include speculative or invented data — use realistic public info`

// 博客文章生成 Prompt
const BLOG_PROMPT = (topic) => `You are the lead content editor at FindInChina, a curated platform for verified Chinese suppliers.

Write an 800-word English blog article about: "${topic}"

Style: editorial, data-driven, helpful for global buyers sourcing from China. Include real brand examples with approximate prices, concrete facts, actionable advice. No fluff. Use markdown with # title, ## sections, - lists. End with a Sources section.`

async function generateReport(category, brands) {
  const completion = await client.chat.completions.create({
    model: 'deepseek-v4-flash',
    messages: [
      { role: 'system', content: 'You are a meticulous brand researcher producing structured JSON data.' },
      { role: 'user', content: REPORT_PROMPT(category, brands) },
    ],
    temperature: 0.5,
    max_tokens: 6000,
    response_format: { type: 'json_object' },
  })
  const raw = completion.choices[0].message.content || ''
  try {
    return JSON.parse(raw)
  } catch (e) {
    // Try to extract JSON from markdown fences
    const match = raw.match(/```json\n([\s\S]*?)```/)
    if (match) return JSON.parse(match[1])
    console.error('JSON 解析失败，原始输出前500字:', raw.slice(0, 500))
    return null
  }
}

async function generateBlog(topic) {
  const completion = await client.chat.completions.create({
    model: 'deepseek-v4-flash',
    messages: [
      { role: 'system', content: 'You are the lead content editor at FindInChina.' },
      { role: 'user', content: BLOG_PROMPT(topic) },
    ],
    temperature: 0.7,
    max_tokens: 3000,
  })
  return completion.choices[0].message.content || ''
}

async function main() {
  if (!fs.existsSync(OUT_DIR)) fs.mkdirSync(OUT_DIR, { recursive: true })

  // 第一批报告：3 个新品类
  const reports = [
    { category: 'Outdoor gear', brands: ['Naturehike', 'TETON Sports (OEM)', 'DECATHLON (OEM)', 'KingCamp', 'FIELDOOR'] },
    { category: 'GaN chargers', brands: ['CUKTECH', 'PISEN', 'Baseus', 'UGREEN', 'Anker (OEM)'] },
    { category: 'Mechanical keyboards', brands: ['Keychron', 'Epomaker', 'Akko', 'Royal Kludge', 'Aula'] },
  ]

  const summary = { generated_at: new Date().toISOString(), reports: [], blogs: [] }

  for (const r of reports) {
    console.log(`生成报告: ${r.category}...`)
    const data = await generateReport(r.category, r.brands)
    if (data) {
      const file = path.join(OUT_DIR, `report-${data.slug || r.category.toLowerCase().replace(/\s+/g, '-')}.json`)
      fs.writeFileSync(file, JSON.stringify(data, null, 2))
      summary.reports.push({ category: r.category, slug: data.slug, title: data.title, file })
      console.log(`  ✅ ${data.title}`)
    } else {
      summary.reports.push({ category: r.category, error: '生成失败' })
    }
    await new Promise(res => setTimeout(res, 2000))
  }

  // 第一批博客：2 篇
  const blogTopics = [
    'How to verify a Chinese factory is real before you send a deposit (2026 guide)',
    'Made in China 2026: 15 underrated Chinese tech brands worth buying directly',
  ]
  for (const topic of blogTopics) {
    console.log(`生成博客: ${topic.slice(0, 50)}...`)
    const content = await generateBlog(topic)
    if (content) {
      const slug = topic.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/^-|-$/g, '').slice(0, 80)
      const file = path.join(OUT_DIR, `blog-${slug}.md`)
      fs.writeFileSync(file, content)
      summary.blogs.push({ topic, file })
      console.log(`  ✅ ${file}`)
    }
    await new Promise(res => setTimeout(res, 2000))
  }

  fs.writeFileSync(path.join(OUT_DIR, 'summary.json'), JSON.stringify(summary, null, 2))
  console.log('\n=== 全部完成 ===')
  console.log(`输出目录: ${OUT_DIR}`)
  console.log(`报告 ${summary.reports.length} 篇, 博客 ${summary.blogs.length} 篇`)
}

main().catch(e => { console.error('执行失败:', e.message); process.exit(1) })
